### PR TITLE
Readding IfIndex in display and name description as suggested by Matt

### PR DESCRIPTION
**Pull request by ViDou83**
Readding IfIndex in display and name description following our internal conversation

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -373,28 +373,28 @@ void WriteInterfaces()
...
                 );
             } else {
-                printf("IF: medium=eth\tID=%u\tIfIndex=%u\tVlanID=%i", Interface->PcapNgIfIndex, Interface->LowerIfIndex, Interface->VlanId);
+                printf("IF: medium=eth\t\t\tID=%u\tIfIndex=%u\tVlanID=%i", Interface->PcapNgIfIndex, Interface->LowerIfIndex, Interface->VlanId);
                 StringCchPrintfA(IfName, IF_STRING_MAX_SIZE, "eth:%lu:%i", Interface->LowerIfIndex, Interface->VlanId);
             }
             break;
         case PCAPNG_LINKTYPE_IEEE802_11:
-            printf("IF: medium=wifi ID=%u\tIfIndex=%u", Interface->PcapNgIfIndex, Interface->LowerIfIndex);
+            printf("IF: medium=mbb\t\t\tID=%u\tIfIndex=%u", Interface->PcapNgIfIndex, Interface->LowerIfIndex);
```
Comment (maolson-msft) [inline src/main.c:397]: ?

File: src/main.c [hunk 2]
```diff
@@ -373,28 +373,28 @@ void WriteInterfaces()
...
+                    "vNIC:%s:%s:%lu:%i",
                     Interface->VMNic.SourceNicType,
-                    Interface->VMNic.SourcePortId,
+                    Interface->VMNic.SourcePortName,
+                    Interface->LowerIfIndex,
                     Interface->VlanId
                 );
             } else {
-                printf("IF: medium=eth\tID=%u\tIfIndex=%u\tVlanID=%i", Interface->PcapNgIfIndex, Interface->LowerIfIndex, Interface->VlanId);
+                printf("IF: medium=eth\t\t\tID=%u\tIfIndex=%u\tVlanID=%i", Interface->PcapNgIfIndex, Interface->LowerIfIndex, Interface->VlanId);
```
Comment (maolson-msft) [inline src/main.c:392]: Why was this tab changed to three tabs? Can you share the before and after appearance?

File: src/main.c [hunk 3]
```diff
@@ -373,28 +373,28 @@ void WriteInterfaces()
...
             if (Interface->IsVMNic) {
-                printf("IF: medium=%s\tID=%u\tIfIndex=%u\tVlanID=%i",
+                printf("IF: medium=vNIC-%s\tID=%u\tIfIndex=%u\tVlanID=%i",
                     Interface->VMNic.SourceNicType,
                     Interface->PcapNgIfIndex,
-                    Interface->VMNic.SourcePortId,
+                    Interface->LowerIfIndex,
                     Interface->VlanId
                 );
                 StringCchPrintfA(
```
Comment (maolson-msft) [inline src/main.c:382]: Maybe I missed this in the last review; why is there a printf followed immediately by a StringCchPrintfA here? Why isn't this just one printf call?

File: src/main.c (hunk 1 above)
Comment (ViDou83) [reply src/main.c:397]: This is to have to output aligned as the VMSwitch decoding will produce a much longer interface name... See below

File: src/main.c (hunk 2 above)
Comment (ViDou83) [reply src/main.c:392]: This is to align output as the VMSwitch interface name are longer (prefix vNIC-XXXX)

After :
IF: medium=eth                         ID=0    IfIndex=3       VlanID=0
IF: medium=eth                         ID=1    IfIndex=5       VlanID=0
IF: medium=vNIC-External        ID=2    IfIndex=28      VlanID=98       (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=3    IfIndex=28      VlanID=399      (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=4    IfIndex=28      VlanID=298      (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=5    IfIndex=28      VlanID=397      (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=6    IfIndex=28      VlanID=97       (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=7    IfIndex=28      VlanID=300      (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=8    IfIndex=28      VlanID=100      (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=9    IfIndex=28      VlanID=0        (LWF over IfIndex 6)
IF: medium=vNIC-External        ID=10   IfIndex=28      VlanID=197      (LWF over IfIndex 6)

Before : 
IF: medium=eth  ID=0    IfIndex=3       VlanID=0
IF: medium=eth  ID=1    IfIndex=5       VlanID=0
IF: medium=External     ID=2    IfIndex=1       VlanID=98       (LWF over IfIndex 6)
IF: medium=External     ID=3    IfIndex=1       VlanID=399      (LWF over IfIndex 6)
IF: medium=External     ID=4    IfIndex=1       VlanID=298      (LWF over IfIndex 6)
IF: medium=External     ID=5    IfIndex=1       VlanID=397      (LWF over IfIndex 6)
IF: medium=External     ID=6    IfIndex=1       VlanID=97       (LWF over IfIndex 6)
IF: medium=External     ID=7    IfIndex=1       VlanID=300      (LWF over IfIndex 6)
IF: medium=External     ID=8    IfIndex=1       VlanID=100      (LWF over IfIndex 6)
IF: medium=External     ID=9    IfIndex=1       VlanID=0        (LWF over IfIndex 6)
IF: medium=External     ID=10   IfIndex=1       VlanID=197      (LWF over IfIndex 6)
IF: medium=External     ID=11   IfIndex=1       VlanID=398      (LWF over IfIndex 6)
IF: medium=Synthetic    ID=12   IfIndex=2       VlanID=0        (LWF over IfIndex 6)
IF: medium=External     ID=13   IfIndex=1       VlanID=297      (LWF over IfIndex 6)

File: src/main.c (hunk 3 above)
Comment (ViDou83) [reply src/main.c:382]: It was already present, the first printf is displaying interface information to the console output (see above). The StringCchPrintfA is formatting the interface name, in a more compacted manner, which will be visible in Wireshark (call PcapNgWriteInterfaceDesc).

File: src/main.c (hunk 1 above)
Comment (maolson-msft) [reply src/main.c:397]: I'm asking here why you are printing wifi interfaces as "mbb" (mobile broadband) interfaces. Typo?

File: src/main.c (hunk 3 above)
Comment (maolson-msft) [reply src/main.c:382]: Oh! I'm sorry, I saw "printf" and thought that meant the function prints to the console, but it actually writes to the IfName string.

File: src/main.c (hunk 2 above)
Comment (maolson-msft) [reply src/main.c:392]: OK.

Comment (ViDou83) [reply src/main.c:392]: I can see this is not aligned in this window, but well in the console output!

File: src/main.c (hunk 1 above)
Comment (ViDou83) [reply src/main.c:397]: well spot, this is indeed a typo...

Comment (ViDou83) [reply src/main.c]: See new commit

**Review by maolson-msft — APPROVED** [review]

**Pull request merged**